Dockerfile

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:jammy
+
+MAINTAINER Joey Yandle
+
+RUN apt-get update
+RUN apt-get install -y curl wget unzip pkg-config gcc automake autoconf autotools-dev libtool man gdb git screen sudo rsync cargo
+
+RUN cargo install --locked cargo-lambda
+
+RUN wget -qO- https://get.pnpm.io/install.sh | bash -
+
+RUN mkdir -p /tmp/smithy-install/smithy
+RUN curl -L https://github.com/smithy-lang/smithy/releases/download/1.49.0/smithy-cli-linux-aarch64.zip -o /tmp/smithy-install/smithy-cli-linux-aarch64.zip
+RUN unzip -qo /tmp/smithy-install/smithy-cli-linux-aarch64.zip -d /tmp/smithy-install
+RUN mv /tmp/smithy-install/smithy-cli-linux-aarch64/* /tmp/smithy-install/smithy
+RUN /tmp/smithy-install/smithy/install
+RUN rm -rf /tmp/smithy-install
+
+RUN mkdir -p /tmp/java-install
+RUN curl -L https://download.oracle.com/java/21/latest/jdk-21_linux-aarch64_bin.tar.gz -o /tmp/java-install/jdk-21_linux-aarch64_bin.tar.gz
+
+RUN apt-get install -y protobuf-compiler
+
+RUN apt-get install -y libssl-dev
+RUN cargo install sqlx-cli
+
+RUN apt-get install -y make
+
+RUN mkdir -p /tmp/node-install
+RUN curl -fsSL https://deb.nodesource.com/setup_22.x -o /tmp/node-install/nodesource_setup.sh
+RUN bash /tmp/node-install/nodesource_setup.sh
+RUN apt-get -y install nodejs
+
+RUN mkdir -p /usr/local/java
+RUN cd /usr/local/java && tar xvzf /tmp/java-install/jdk-21_linux-aarch64_bin.tar.gz
+ENV PATH="${PATH}:/usr/local/java/jdk-21.0.3/bin"
+
+RUN mkdir -p /tmp/rustup-install
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -o /tmp/rustup-install/rustup-install.sh
+RUN chmod +x /tmp/rustup-install/rustup-install.sh
+RUN /tmp/rustup-install/rustup-install.sh -y
```

Comment:
Is this necessary if cargo is already installed?

Reply:
It's necessary to be able to get the correct version of `rustc`.

Reply:
Really? That's surprising, I thought it would auto-update to the correct version after running `cargo build`.

Reply:
It was complaining about having `rustc` `1.75` but needing `1.76` or more recent.

Reply:
Oh right, that's because apt-get doesn't install `rustup`, which I suppose cargo delegates to when it notices that it needs a different version of rust. Okay, never mind.

Reply:
I prefer the all in one `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh -s -- -y` so just want to make sure you considered it and are choosing a different path. I prefer this because there's no artifact; we aren't checking any hash here so we don't really need one.